mopidy_mpd/protocol/connection.py

```diff
@@ -51,3 +52,54 @@ def ping(context):
         Does nothing but return ``OK``.
     """
     pass
+
+
+@protocol.commands.add("tagtypes")
+def tagtypes(context, *args):
+    """
+    *mpd.readthedocs.io, connection settings section:*
+
+        ``tagtypes``
+
+        Shows a list of available song metadata.
+
+        ``tagtypes disable {NAME...}``
+
+        Remove one or more tags from the list of tag types the client is interested in.
+
+        ``tagtypes enable {NAME...}``
+
+        Re-enable one or more tags from the list of tag types for this client.
+
+        ``tagtypes clear``
+
+        Clear the list of tag types this client is interested in.
+
+        ``tagtypes all``
+
+        Announce that this client is interested in all tag types.
+    """
+    if not args:
+        pass
+    elif args[0] == "all":
+        context.session.tagtypes = tagtype_list.TAGTYPE_LIST[:]
+    elif args[0] == "clear":
+        context.session.tagtypes.clear()
+    elif args[0] == "disable":
+        context.session.tagtypes = [
+            value for value in context.session.tagtypes if value not in args[1:]
+        ]
+    elif args[0] == "enable":
+        enabled_types = [
+            value for value in args[1:] if value not in context.session.tagtypes
+        ]
+        context.session.tagtypes.extend(enabled_types)
+    else:
+        raise exceptions.MpdArgError("Unknown sub command")
+
+    tagtypes = [
+        value
+        for value in context.session.tagtypes
+        if value in tagtype_list.TAGTYPE_LIST
```

Comment:
Invalid tag types are just ignored. The protocol docs didn't seem to specify what should happen for invalid tag types so hopefully that is ok?

Reply:
Just to clarify as I think you already fixed this. Since the MPD protocol isn't thoroughly defined, in general we just try to do the same as the latest mpd does.